tools/tpm2_clearcontrol.c

```diff
@@ -44,16 +44,63 @@ static bool clearcontrol(ESYS_CONTEXT *ectx) {
     return true;
 }
 
+static bool set_clearcontrol_auth_handle(const char *value) {
```

Comment:
@williamcroberts tpm2_hierarchy_from_optarg does not return ESYS_TR.

Reply:
also it cannot return lockout handle and rightly so since it is not a hierarchy. See #1510

Reply:
Add a flag to tpm2_hierarchy_flags to return the ESAPI handle constant. Probably change TPMI_RH_PROVISION to something generic. Everything is a UINT32 so it will fit in there.

For example modify each segment to do something like this:
```C
    bool is_o = !strcmp(value, "o");
    if (is_o) {
        if (!(flags & TPM2_HIERARCHY_FLAGS_O)) {
            LOG_ERR("Owner hierarchy not supported by this command.");
            return false;
        }
        *hierarchy = !(flags & TPM2_HIERARCHY_FLAGS_ESAPI) ? TPM2_RH_OWNER : ESYS_TR_RH_OWNER;
        return true;
    }
```

Reply:
@williamcroberts since lockout is a permanent handle and not a hierarchy it would not belong in the routine: tpm2_hierarchy_from_optarg. 

Additionally, The issue i am alluding to in #1510 is that https://github.com/tpm2-software/tpm2-tools/blob/5b96ac3983407fa273b4455a091a399ee15f6b48/lib/tpm2_hierarchy.c#L78 does not do all the checks necessary to convert a hex number to one of hierarchy handles owner, platform, endorsement or null.

Reply:
The goal here is to have consistent behavior between tools including things like error messages. So unfortunately not using a common routine for changing persistent handles can cause differences in behavior. Since this is just for platform or lockout, I guess its fine for now.